drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/GuidedDecisionTableModelVisitor.java

```diff
@@ -89,7 +89,7 @@ private String convertToFullyQualifiedClassName(final String factType) {
         }
         String fullyQualifiedClassName = null;
         for (Import imp : imports.getImports()) {
-            if (imp.getType().endsWith(factType)) {
+            if (imp.getType().endsWith("." + factType)) {
```

Comment:
Hmm I think this is still not proper fix, imagine a testcase:
```java
    @Test
    public void differentPackagesSameNames() {
        final GuidedDecisionTable52 model = new GuidedDecisionTable52();
        model.getImports().addImport(new Import("org.test.demo.Person"));
        model.getImports().addImport(new Import("org.test.Person"));
        final Pattern52 pattern52 = mock(Pattern52.class);
        doReturn("Person").when(pattern52).getFactType();

        model.getConditions().add(pattern52);

        final Set<String> consumedModelClasses = new GuidedDecisionTableModelVisitor(model).getConsumedModelClasses();

        assertTrue(consumedModelClasses.contains("org.test.Person"));
    }
```

Reply:
At the moment the guided rule editors do not support the same class name from two sources. This is because the generated DRL imports every import and then in the actual rule part you only have Person. The compiler does not know what to pick.